diff --git a/mkspecs/common/msvc-version.conf b/mkspecs/common/msvc-version.conf
index ccd809abf3..3fb55c9d81 100644
--- a/mkspecs/common/msvc-version.conf
+++ b/mkspecs/common/msvc-version.conf
@@ -100,11 +100,16 @@ greaterThan(QMAKE_MSC_VER, 1909) {
     # API is used in direct2d, but also in multimedia, positioning and sensors.
     # We can try again with a later version of Visual Studio.
     # QMAKE_CXXFLAGS_STRICTCXX = -permissive-
+
     # MSVC partially supports the following, but '__cplusplus' definition is set
     # as for C++98 until MSVC fully conforms with C++14, see
     # https://developercommunity.visualstudio.com/content/problem/139261/msvc-incorrectly-defines-cplusplus.html
-    # QMAKE_CXXFLAGS_CXX14 = -std:c++14
-    # QMAKE_CXXFLAGS_CXX1Z = -std:c++latest
+    # Support became available in MSVC 2017 15.7:
+    greaterThan(QMAKE_MSC_VER, 1913) {
+        QMAKE_CXXFLAGS      += -Zc:__cplusplus
+        QMAKE_CXXFLAGS_CXX14 = -std:c++14
+        QMAKE_CXXFLAGS_CXX1Z = -std:c++17
+    }
 }
 
 greaterThan(QMAKE_MSC_VER, 1910) {
